Xamarin.Forms.Platform.iOS/CollectionView/TemplatedCell.cs

```diff
@@ -25,18 +27,27 @@ protected TemplatedCell(CGRect frame) : base(frame)
 
 		internal IVisualElementRenderer VisualElementRenderer { get; private set; }
 
+		public override void ConstrainTo(CGSize constraint)
+		{
+			ConstrainedSize = constraint;
+		}
+
 		public override void ConstrainTo(nfloat constant)
 		{
 			ConstrainedDimension = constant;
+
+			// Reset constrained size in case ItemSizingStrategy changes
+			// and we want to measure each item
+			ConstrainedSize = default(CGSize);
 		}
 
 		public override UICollectionViewLayoutAttributes PreferredLayoutAttributesFittingAttributes(
 			UICollectionViewLayoutAttributes layoutAttributes)
 		{
 			var preferredAttributes = base.PreferredLayoutAttributesFittingAttributes(layoutAttributes);
 
-			// Measure this cell (including the Forms element)
-			var size = Measure();
+			// Measure this cell (including the Forms element) if there is no constrained size
+			var	size = ConstrainedSize == default(CGSize) ? Measure() : ConstrainedSize;
```

Comment:
The reason `MeasureFirstItem` was broken is we were measuring here and assigning a new size to the cell. Since `MeasureFirstItem` constrains us to the first item, I added `ConstrainedSize` which is passed down from `PrepareCellForLayout()` in `ItemsViewLayout` as `ItemSize`.